src/Aspire.Dashboard/Components/Layout/MainLayout.razor.cs

```diff
@@ -76,6 +78,10 @@ protected override async Task OnInitializedAsync()
                 var newValue = ThemeManager.Theme!;
 
                 await _jsModule.InvokeVoidAsync("updateTheme", newValue);
+                var effectiveTheme = await _jsModule.InvokeAsync<string>("getCurrentTheme");
+                ThemeManager.EffectiveTheme = effectiveTheme;
+
+                await InvokeAsync(StateHasChanged);
```

Comment:
Be careful with calling StateHasChanged and minimize how often it is used. I think in this case, it could be limited to just the new text dialog. Listening to theme change and calling state change in the dialog would be better than doing it globally.